src/main/java/graphql/annotations/EnhancedExecutionStrategy.java

```diff
@@ -55,20 +63,45 @@ protected ExecutionResult resolveField(ExecutionContext executionContext, GraphQ
                 clientMutationId = clientMutationIdVal.getValue();
             }
 
-            return completeValue(executionContext, fieldDef.getType(), fields, clientMutationId);
+            TypeInfo fieldTypeInfo = newTypeInfo(fieldDef.getType(), parameters.typeInfo());
+            ExecutionParameters newParameters = newParameters()
```

Comment:
FYI this code will fail once next 3.x is released because in current master it has one more parameter - null validator.

I implemented the change to mitigate that ( https://github.com/graphql-java/graphql-java/pull/479 ), but it means that you have to lock your release to 3.0.0 and make an additional release (backward incompatible) for 3.x once it's out